sections/docker/Caching-for-shorter-build-time.md

```diff
@@ -0,0 +1,120 @@
+# Leverage caching to reduce build times
+
+## One paragraph explainer
+
+Docker images are a combination of layers, each instruction in your Dockerfile creates a layer. The docker daemon can reuse those layers between builds if the instructions are identicals or in the case of a `COPY` or `ADD` files used are identical.
+It is important to layout your Dockerfile correctly to reduce the number of moving parts in your build, the less updated instructions should be at the top and the ones constantly changing (like app code) should be at the bottom. 
+Rebuilding a whole docker image from cache can be nearly instantaneous if done correctly.
+
+⚠️ If the cache can't be used for a particular layer, all the subsquent layers will be invalidated too. That's why order is important.
```

Comment:
Consider putting this great sentence somewhere at the opening of the text, it will help to understand what you wrote about the layers